lib/Dialect/TritonGPU/Transforms/RemoveLayoutConversions.cpp

```diff
@@ -137,11 +137,12 @@ class SimplifyReduceCvt : public mlir::RewritePattern {
           op->getLoc(), newTy, newOperands[i]);
     }
 
+    rewriter.setInsertionPoint(reduce);
     auto newReduce = rewriter.create<triton::ReduceOp>(
         op->getLoc(), newOperands, reduce.getAxis());
     auto &newCombineOp = newReduce.getCombineOp();
-    rewriter.inlineRegionBefore(reduce.getCombineOp(), newCombineOp,
-                                newCombineOp.end());
+    rewriter.cloneRegionBefore(reduce.getCombineOp(), newCombineOp,
```

Comment:
Inlining is destructive, so if the rewrite produces an invalid output this would break the original `ReduceOp`'s region.